cognite/powerops/client/shop_results_compare.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+import logging
+from datetime import datetime
+from typing import TYPE_CHECKING, Optional, Sequence
+
+import matplotlib.pyplot as plt
+
+from cognite.powerops.client.shop_result_files import ShopYamlFile
+from cognite.powerops.utils.plotting import ax_plot_time_time_series
+
+if TYPE_CHECKING:
+    from cognite.powerops import PowerOpsClient
+
+logger = logging.getLogger(__name__)
+
+
+class ShopResultsCompareAPI:
+    def __init__(self, po_client: PowerOpsClient):
+        self._po_client = po_client
+
+    def time_series_plots(
+        self, post_run_list: Sequence[ShopYamlFile], comparison_key: str, labels: Optional[Sequence[str]] = ()
+    ):
+        """Stacked line charts of the given post runs at the same key.
+        Labels must be in the same order as the post runs."""
+        if labels and (len(labels) != len(set(labels)) or len(labels) != len(post_run_list)):
+            logger.error("Titles mus be unique and match the number of provided post runs")
+            return
+
+        plots: dict[str, dict[datetime, float]] = {}
+
+        for i, shop_yaml in enumerate(post_run_list):
+            if time_series := shop_yaml._retrieve_time_series_dict(comparison_key):
+                plot_label = labels[i] if labels else shop_yaml.name
+                plots[plot_label] = time_series
+
+        fig, ax = plt.subplots(figsize=(10, 10))
+        fig.autofmt_xdate()
+        for label, time_series in plots.items():
+            ax_plot_time_time_series(ax, time_series, label)
```

Comment:
You could consider returning the object to be plotted from this function to make what happens in this functions more transparent. Could maybe also move line 38 and 39 into that same function and rename it to: 
create_time_series_plot which returns an Axes object.